src/promptflow-azure/promptflow/azure/_storage/blob/client.py

```diff
@@ -63,29 +66,32 @@ def get_datastore_container_client(
 
 def get_default_datastore(
     subscription_id: str, resource_group_name: str, workspace_name: str, credential: Optional[object]
-) -> Datastore:
+) -> Tuple[Datastore, str]:
 
     datastore_key = _get_datastore_client_key(subscription_id, resource_group_name, workspace_name)
-    datastore = _get_datastore_from_cache(datastore_key=datastore_key)
+    datastore, datastore_credential = _get_datastore_from_cache(datastore_key=datastore_key)
     if datastore is None:
         with _thread_lock:
-            datastore = _get_datastore_from_cache(datastore_key=datastore_key)
+            datastore, datastore_credential = _get_datastore_from_cache(datastore_key=datastore_key)
             if datastore is None:
-                datastore = _get_default_datastore(subscription_id, resource_group_name, workspace_name, credential)
+                datastore, datastore_credential = _get_default_datastore(
```

Comment:
Is `datastore_definition` a better name to understand?
Since we have datastore and datastore_credential.

Reply:
Fixed.